launch_ros/launch_ros/actions/load_composable_nodes.py

```diff
@@ -0,0 +1,171 @@
+        if composable_node_description.remappings is not None:
+            for from_, to in composable_node_description.remappings:
+                request.remap_rules.append('{}:={}'.format(
+                    perform_substitutions(context, list(from_)),
+                    perform_substitutions(context, list(to)),
+                ))
+        if composable_node_description.parameters is not None:
+            request.parameters = [
+                param.to_parameter_msg() for param in to_parameters_list(
+                    context, evaluate_parameters(
+                        context, composable_node_description.parameters
+                    )
+                )
+            ]
+        if composable_node_description.extra_arguments is not None:
+            request.extra_arguments = [
+                param.to_parameter_msg() for param in to_parameters_list(
+                    context, evaluate_parameters(
+                        context, composable_node_description.extra_arguments
+                    )
+                )
+            ]
+        response = self.__rclpy_load_node_client.call(request)
+        if not response.success:
+            self.__logger.error("Failed to load node '{}' in container '{}': {}".format(
+                response.full_node_name, self.__target_container.node_name, response.error_message
+            ))
+
+    def _load_in_sequence(
+        self,
+        composable_node_descriptions: List[ComposableNode],
+        context: LaunchContext
+    ) -> None:
+        """
+        Load composable nodes sequentially.
+
+        :param composable_node_descriptions: descriptions of composable nodes to be loaded
+        :param context: current launch context
+        """
+        next_composable_node_description = composable_node_descriptions[0]
+        composable_node_descriptions = composable_node_descriptions[1:]
+        self._load_node(next_composable_node_description, context)
+        if len(composable_node_descriptions) > 0:
+            context.add_completion_future(
+                context.asyncio_loop.run_in_executor(
+                    None, self._load_in_sequence, composable_node_descriptions, context
+                )
+            )
+
+    def _on_container_start(
+        self,
+        event: ProcessStarted,
+        context: LaunchContext
+    ) -> Optional[List[Action]]:
+        """Load nodes on container process start."""
+        self._load_in_sequence(self.__composable_node_descriptions, context)
+        return None
+
+    def execute(
+        self,
+        context: LaunchContext
+    ) -> Optional[List[Action]]:
+        """Execute the action."""
+        # Create a client to load nodes in the target container.
+        self.__rclpy_load_node_client = context.locals.launch_ros_node.create_client(
+            composition_interfaces.srv.LoadNode, '{}/_container/load_node'.format(
+                self.__target_container.node_name
```

Comment:
I think `~/_container/load_node` will expand to `/name/space/node_name/_container/load_node`
This means the `ComposableNodeContainer` must know the node name and namespace. Right now a `ComposableNodeContainer` can be created without specifying these, resulting in `/<node_name_unspecified>/_container/load_node` by default.  

https://github.com/ros2/launch_ros/blob/1340fc8139749b570a43109561c36623c16cbb58/launch_ros/launch_ros/actions/node.py#L171-L172

This isn't a valid service name, so it fails with

```
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Invalid service name: topic name must not contain characters other than alphanumerics, '_', '~', '{', or '}':
```

One solution might be to make `node_name` and `node_namespace` mandatory arguments to `ComposableNodeContainer`'s `__init__()`.

Reply:
That's a **very** good point. Fixed as suggested.